app/src/main/java/de/tum/in/tumcampusapp/widgets/MVVWidgetService.java

```diff
@@ -27,10 +27,12 @@ public RemoteViewsFactory onGetViewFactory(Intent intent) {
         private final Context applicationContext;
         private List<TransportManager.Departure> departures = new ArrayList<>();
         private int appWidgetID;
+        private boolean forceLoadDepartures = false;
 
         MVVRemoteViewFactory(Context applicationContext, Intent intent) {
             this.applicationContext = applicationContext.getApplicationContext();
             this.appWidgetID = intent.getIntExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, -1);
+            this.forceLoadDepartures = intent.getBooleanExtra(MVVWidget.MVV_WIDGET_FORCE_RELOAD, false);
```

Comment:
As I do not know this: when is this contructor called? Does it rerun once the setting has been updated?